aten/src/ATen/cuda/CUDAApplyUtils.cuh

```diff
@@ -198,8 +199,8 @@ inline void rearrangeDims(detail::TensorInfo<T1, IndexType>* aInfo,
 
 // Threads per block for our apply kernel
 // FIXME: use occupancy calculator instead
-#define AT_APPLY_THREADS_PER_BLOCK 32 * 16
-#define AT_APPLY_BLOCKS_PER_SM 4
+#define AT_APPLY_THREADS_PER_BLOCK C10_MAX_THREADS_PER_BLOCK(512)
+#define AT_APPLY_BLOCKS_PER_SM C10_MIN_BLOCKS_PER_SM(512, 4)
```

Comment:
I just want to confirm here: does `__launch_bounds__` support being passed constexpr expressions? If so, since we have the patient open, I'd also like to de-macro-ify this. (If it doesn't, I guess we ought not... but it seems that it is able to handle arithmetic expressions, so... wanted to check.)

Reply:
Let me check

Reply:
Confirming that launch bounds can handle constexpr.